### PR TITLE
Migrate the example and docker tests to GitHub Actions

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Migrate the example and docker tests to GitHub Actions

## How is this patch tested?

GitHub Actions

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM continuumio/miniconda:4.5.4
 
+
```
Comment (harupy) [inline examples/docker/Dockerfile:3]: Will revert this change after veryfing that the example and docker tests pass

**Comment by harupy** [comment]
Once this PR get merged, I'll open another PR to remove files and options that are related to Travis.

File: Dockerfile [hunk 2]
```diff
@@ -10,10 +10,10 @@ RUN apt-get update && \
 # cmake and protobuf-compiler required for onnx install
     cmake protobuf-compiler &&  \
 # install required python packages
-    pip install -r dev-requirements.txt && \
-    pip install -r test-requirements.txt && \
+    pip install -r dev-requirements.txt --no-cache-dir && \
+    pip install -r test-requirements.txt --no-cache-dir && \
 # install mlflow in editable form
-    pip install -e . && \
+    pip install --no-cache-dir -e . && \
```
Comment (harupy) [inline Dockerfile:16]: Fix for https://github.com/mlflow/mlflow/runs/872426168#step:5:1884

```
...
Find out more about deployment here:

  https://bit.ly/CRA-deploy

Error processing tar file(exit status 1): mkdir /root/.npm/_cacache/index-v5/fc: no space left on device
##[error]Process completed with exit code 1.
```

File: mlflow/server/js/package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
   },
   "scripts": {
     "start": "react-app-rewired start",
-    "build": "react-app-rewired --max_old_space_size=4096 build",
+    "build": "react-app-rewired --max_old_space_size=8192 build",
```
Comment (harupy) [inline mlflow/server/js/package.json:76]: Fix for https://github.com/mlflow/mlflow/runs/872050285#step:5:1863

```
...
<undefined>,chunk=0x0aa42e751361 <String[2]: n[>,middleMapping=0x3320f96159a1 <Object map = 0x38e86e00a081>)
    2: SourceNode_walk [0x5dd5b275769] [/app/mlflow/server/js/node_modu...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x8fb090 node::Abort() [node]
 2: 0x8fb0dc  [node]
```

**Comment by smurching** [comment]
Cool! I see both docker & examples tests were triggered & passed in https://github.com/mlflow/mlflow/runs/872630445. It's surprising that python large tests [took 30 min on the most-recent commit](https://github.com/mlflow/mlflow/runs/876038300), but the only difference between the most-recent commit and the previous one (where the [tests took just 11 min](https://github.com/mlflow/mlflow/runs/875741674])) was the addition of a status badge, so I don't think any code change in this PR is responsible for the variance - let's keep an eye on it.

**Review by smurching — APPROVED** [review]
LGTM
